api/src/main/java/io/druid/data/input/impl/JavaScriptParseSpec.java

```diff
@@ -45,6 +48,7 @@ public JavaScriptParseSpec(
   )
   {
     super(timestampSpec, dimensionsSpec);
+    Preconditions.checkState(config.isEnabled(), "JavaScript is disabled");
```

Comment:
The original idea behind delaying these checks until actual usage is that when you disable JavaScript, you might still want your Druid nodes to be able to deserialize JavaScript-based objects. For example, maybe you had used JavaScript aggregators in your ingest tasks, and after disabling JavaScript you still want to be able to view old completed tasks from the task table in the metadata store.

So I think it's worth being able to create these objects even if JavaScript is disabled, just make sure the JavaScript is never actually compiled.

Reply:
Thanks. Changed to postpone the JavaScript configuration check.